projects/libs/core-library-typescript/src/common-library/startup-check-runner.ts

```diff
@@ -0,0 +1,52 @@
+import { StartupCheck } from "./startup-check";
+import { StartupStatus, STARTUP_STATUS } from "./startup-status";
+import { logger } from "../common-library";
+
+export class StartupCheckRunner {
+  name: string;
+  startupCheckers: StartupCheck[] = [];
+
+  constructor(name: string, startupCheckers: StartupCheck[]) {
+    this.name = name;
+    this.startupCheckers = startupCheckers;
+  }
+
+  run(): Promise<StartupStatus> {
+
+    const startupCheckPromises = this.startupCheckers.map((s: StartupCheck) => s.startupCheck());
+    return Promise.all(startupCheckPromises)
+      .then(this.mergeStartupStatuses)
+      .catch(err => {
+        logger.error(err);
+        return <StartupStatus>{ status: STARTUP_STATUS.FAIL, message: err.message };
+      });
+  }
+
+  private mergeStartupStatuses = (startupStatuses: StartupStatus[]): StartupStatus => {
+
+    let hasAtLeastOneOk = false;
+    let hasAtLeastOneFailure = false;
```

Comment:
this is a general note, typescript loves functional programming 'style', and when you use let by definition you will probably do something that is not functional by nature (sometimes its a good idea, usually not)